### PR TITLE
Leave feedback

### PR DESCRIPTION
**Pull request by IPoyda**
Nice header. Working add feedback on user and training.

### CODE REVIEW

File: frontend/app/scripts/app.js [hunk 1]
```diff
@@ -17,7 +17,11 @@ angular
     'ngSanitize',
     'ngTouch',
     'smart-table',
-    'ngStorage'
+    'ngStorage',
+    'ngDialog',
+    'ngDialog',
+    'ui.bootstrap',
```
Comment (pashapovalov) [inline frontend/app/scripts/app.js]: Added twice

File: frontend/app/scripts/app.js [hunk 2]
```diff
@@ -17,7 +17,11 @@ angular
     'ngSanitize',
     'ngTouch',
     'smart-table',
-    'ngStorage'
+    'ngStorage',
+    'ngDialog',
+    'ngDialog',
```
Comment (pashapovalov) [inline frontend/app/scripts/app.js]: Added twice

File: frontend/app/scripts/controllers/popups/leaveFeddbackCtrl.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    ];
+
+    $scope.leaveTrainingFeedback = function() {
+      var userId =1;
+      var trainingId = 123;
+      var feed = new TrainingFeedbackService();
+      feed.id = trainingId;
+      feed.userId = userId;
+      feed.effectiveness = $scope.$$childHead.ratings.current;        //плохо
+      feed.understandable = $scope.understandable;
```
Comment (pashapovalov) [inline frontend/app/scripts/controllers/popups/leaveFeddbackCtrl.js]: Please in code use ONLY english ;)

File: frontend/app/scripts/controllers/popups/leaveFeddbackCtrl.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      { grade: 1},
+      { grade: 2},
+      { grade: 3},
+      { grade: 4}
+    ];
+
+    $scope.leaveTrainingFeedback = function() {
+      var userId =1;
+      var trainingId = 123;
+      var feed = new TrainingFeedbackService();
```
Comment (pashapovalov) [inline frontend/app/scripts/controllers/popups/leaveFeddbackCtrl.js]: Please add TODO comment, that it is test data.

File: frontend/app/scripts/controllers/popups/leaveFeddbackCtrl.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+      { grade: 0},
+      { grade: 1},
+      { grade: 2},
+      { grade: 3},
+      { grade: 4}
+    ];
+
+    $scope.leaveTrainingFeedback = function() {
+      var userId =1;
+      var trainingId = 123;
```
Comment (pashapovalov) [inline frontend/app/scripts/controllers/popups/leaveFeddbackCtrl.js:31]: Please add TODO comment, that it is test data.

File: frontend/app/scripts/controllers/popups/leaveFeddbackCtrl.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+      var feed = new TrainingFeedbackService();
+      feed.id = trainingId;
+      feed.userId = userId;
+      feed.effectiveness = $scope.$$childHead.ratings.current;        //плохо
```
Comment (pashapovalov) [inline frontend/app/scripts/controllers/popups/leaveFeddbackCtrl.js]: Please in code use ONLY english ;)

File: frontend/app/scripts/controllers/popups/leaveFeddbackCtrl.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    };
+
+    $scope.leaveUserFeedback = function() {
+      var userId =123; //id of user on we write feedback
+      var feedbackerId = 1; // id of feedbacker
+      var feed = new UserFeedbackService();
+      feed.userId = userId;
+      feed.feedbackerId = feedbackerId;
+      feed.attendanc = $scope.attendance;
+      feed.attitude = $scope.attitude;
```
Comment (pashapovalov) [inline frontend/app/scripts/controllers/popups/leaveFeddbackCtrl.js:56]: If you add to the scope object 'entity' (or 'feedData', or other) and move fields to it. You just can code like this:

``` javascript
_.extend(feed, $scope.entity);
```

File: frontend/app/scripts/controllers/userprofile.js [hunk 8]
```diff
@@ -2,6 +2,17 @@
  * Created by Natsik on 13.07.2015.
  */
 angular.module('frontendApp')
-  .controller('UserProfileCtrl', function () {
-
-  });;
+  .controller('UserProfileCtrl',['$scope','ngDialog', function ($scope,ngDialog) {
```
Comment (pashapovalov) [inline frontend/app/scripts/controllers/userprofile.js:5]: Please, use camelCase for the file name.

File: frontend/app/views/header.html [hunk 9]
```diff
@@ -21,15 +21,27 @@
...
               ng-show="isAdmin()" >
             <a ng-href="#/users">Users List</a></li>
-          <li>
+          <li ng-class="{ active: isActive('traininglist')}">
             <a ng-href="#/traininglist">Training List</a></li>
         </ul>
 
         <ul class="nav navbar-nav navbar-right">
           <li>
+            <div ng-controller="HeaderCtrl" id="content">
```
Comment (pashapovalov) [inline frontend/app/views/header.html]: Please look that this file already contains HeaderCtrl

File: src/main/java/com/exadel/filter/CORSFilter.java [hunk 10]
```diff
@@ -18,7 +18,7 @@ public void doFilter(ServletRequest req, ServletResponse res,
 		response.setHeader("Access-Control-Allow-Origin", "*");
 		response.setHeader("Access-Control-Allow-Methods", "POST, GET, OPTIONS, DELETE");
 		response.setHeader("Access-Control-Max-Age", "3600");
-		response.setHeader("Access-Control-Allow-Headers", "x-requested-with"); //content-type
+		response.setHeader("Access-Control-Allow-Headers", "x-requested-with,accept,content-type"); //content-type
```
Comment (pashapovalov) [inline src/main/java/com/exadel/filter/CORSFilter.java:21]: Are you sure that you need commit this file?

File: frontend/app/views/userprofile.html [hunk 11]
```diff
@@ -1,3 +1,5 @@
 <div ng-controller="UserProfileCtrl as userprofile">
   User Profile panel.
+  <button type="button" ng-click="leaveTrainingFeedback()" ng-dialog-class="ngdialog-theme-plain">SMTH1</button>
```
Comment (pashapovalov) [inline frontend/app/views/userprofile.html]: I understand that this is temporary button, but use more understandable names. Thanks!

File: frontend/app/views/popups/leaveUserFeedback.html [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+    <div class="form-group feedbackBlock" style="margin: 5px">
+      <div class="col-md-8" style="margin-top: 3%">
+        <p>English level</p>
+      </div>
+      <div ng-controller="LeaveFeedbackCtrl" style="margin-top: 3%">
+        <select ng-model="ddlLevle" ng-options="levle.Name as levle.Name for levle in levles"></select>
+      </div>
+    </div>
+
+    <div class="form-group feedbackBlock" style="margin: 5px">
```
Comment (pashapovalov) [inline frontend/app/views/popups/leaveUserFeedback.html]: No style in html!

File: frontend/app/views/popups/leaveTrainingFeedback.html [hunk 13]
```diff
@@ -0,0 +1,107 @@
+<div style="padding: 10px;">
+  <h2 style="text-align: center; margin-top: 10px">Leave feedback on this training</h2>
+</div>
```
Comment (pashapovalov) [inline frontend/app/views/popups/leaveTrainingFeedback.html:3]: No style in html!

File: frontend/app/styles/main.less [hunk 14]
```diff
@@ -89,6 +131,61 @@ footer .footer-container {
...
+.rating {
+  color: #3e3d40;
+  margin: 0;
+  padding: 0;
+}
+ul.rating {
+  display: inline-block;
+}
+.rating li {
+  list-style-type: none;
```
Comment (pashapovalov) [inline frontend/app/styles/main.less]: Please use less style:

``` javascript
.rating {
  color: #3e3d40;
  margin: 0;
  padding: 0;

  li {
     ....
  }

  .field { 
    .....
  }
}
```

File: src/main/java/com/exadel/filter/CORSFilter.java (hunk 10 above)
Comment (IPoyda) [inline src/main/java/com/exadel/filter/CORSFilter.java:21]: yes I am definitely sure)

**Pull request merged**